sdk/eventgrid/azure-messaging-eventgrid-namespaces/src/test/java/com/azure/messaging/eventgrid/namespaces/EventGridClientTests.java

```diff
@@ -0,0 +1,291 @@
+            .setTime(testResourceNamer.now())
+            .setId(testResourceNamer.randomUuid());
+        client.publishCloudEvent(TOPIC_NAME, event, true);
+    }
+
+    @Test
+    void publishBatchOfCloudEventsSync() {
+        EventGridClient client = buildSyncClient();
+        client.publishCloudEvents(TOPIC_NAME, Arrays.asList(getCloudEvent(), getCloudEvent()));
+    }
+
+    @Test
+    void receiveBatchOfCloudEventsSync() {
+        EventGridClient client = buildSyncClient();
+        client.publishCloudEvent(TOPIC_NAME, getCloudEvent());
+        ReceiveResult result = client.receiveCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, 10, Duration.ofSeconds(10));
+        assertNotNull(result);
+        assertFalse(result.getValue().isEmpty());
+    }
+
+    @Test
+    void acknowledgeBatchOfCloudEventsSync() {
+        EventGridClient client = buildSyncClient();
+        client.publishCloudEvent(TOPIC_NAME, getCloudEvent());
+        ReceiveResult receiveResult = client.receiveCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, 1, Duration.ofSeconds(10));
+        AcknowledgeOptions acknowledgeOptions = new AcknowledgeOptions(Arrays.asList(receiveResult.getValue().get(0).getBrokerProperties().getLockToken()));
+        AcknowledgeResult acknowledgeResult = client.acknowledgeCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, acknowledgeOptions);
+        assertNotNull(acknowledgeResult);
+        assertFalse(acknowledgeResult.getSucceededLockTokens().isEmpty());
+    }
+
+    @Test
+    void releaseBatchOfCloudEventsSync() {
+        EventGridClient client = buildSyncClient();
+        client.publishCloudEvent(TOPIC_NAME, getCloudEvent());
+        ReceiveResult receiveResult = client.receiveCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, 1, Duration.ofSeconds(10));
+        ReleaseOptions releaseOptions = new ReleaseOptions(Collections.singletonList(receiveResult.getValue().get(0).getBrokerProperties().getLockToken()));
+        ReleaseResult releaseResult = client.releaseCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, releaseOptions);
+        assertNotNull(releaseResult);
+        assertFalse(releaseResult.getSucceededLockTokens().isEmpty());
+    }
+
+    @Test
+    void releaseBatchOfCloudEventsWithDelaySync() {
+        EventGridClient client = buildSyncClient();
+        client.publishCloudEvent(TOPIC_NAME, getCloudEvent());
+        ReceiveResult receiveResult = client.receiveCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, 1, Duration.ofSeconds(10));
+        ReleaseOptions releaseOptions = new ReleaseOptions(Collections.singletonList(receiveResult.getValue().get(0).getBrokerProperties().getLockToken()));
+        ReleaseResult releaseResult = client.releaseCloudEvents(TOPIC_NAME, EVENT_SUBSCRIPTION_NAME, releaseOptions, ReleaseDelay.BY10_SECONDS);
+        assertNotNull(releaseResult);
+        assertFalse(releaseResult.getSucceededLockTokens().isEmpty());
+    }
+
+    @Test
+    void rejectBatchOfCloudEventsSync() {
```

Comment:
nit: generally, we split sync and async tests into separate .java files. It might be good to split them up here as well just be consistent with the test suites of rest of the libraries.